Starting test execution, please wait...
Logging Vstest Diagnostics in file: C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\fine-code-coverage\diagnostics.log
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:01.62]     HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_all_feedbacks [FAIL]
[xUnit.net 00:00:01.63]     HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_by_id [FAIL]
  Failed HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_all_feedbacks [9 ms]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at HospitalApp.Services.FeedbackService.GetAll() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalApp\Services\FeedbackService.cs:line 25
   at HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_all_feedbacks() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\FeedbackServiceTests.cs:line 25
  Failed HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_by_id [2 ms]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at HospitalApp.Services.FeedbackService.GetById(Int32 feedbackId) in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalApp\Services\FeedbackService.cs:line 32
   at HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_by_id() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\FeedbackServiceTests.cs:line 33
[xUnit.net 00:00:11.37]     HospitalAppTests.ServicesTests.PatientServiceTests.Get_appointment_patient [FAIL]
  Failed HospitalAppTests.ServicesTests.PatientServiceTests.Get_appointment_patient [9 s]
  Error Message:
   System.ArgumentException : An item with the same key has already been added. Key: 22
  Stack Trace:
     at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryTable`1.Create(IUpdateEntry entry)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryStore.ExecuteTransaction(IList`1 entries, IDiagnosticsLogger`1 updateLogger)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(DbContext _, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.Storage.NonRetryingExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges()
   at HospitalAppTests.ServicesTests.PatientServiceTests.CreateDoctors(MyDbContext context) in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 223
   at HospitalAppTests.ServicesTests.PatientServiceTests.Get_appointment_patient() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 81
[xUnit.net 00:00:11.42]     HospitalAppTests.ServicesTests.PatientServiceTests.Set_random_general_practitioner [FAIL]
  Failed HospitalAppTests.ServicesTests.PatientServiceTests.Set_random_general_practitioner [47 ms]
  Error Message:
   Microsoft.EntityFrameworkCore.DbUpdateConcurrencyException : Attempted to update or delete an entity that does not exist in the store.
  Stack Trace:
     at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryTable`1.Delete(IUpdateEntry entry)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryStore.ExecuteTransaction(IList`1 entries, IDiagnosticsLogger`1 updateLogger)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(DbContext _, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.Storage.NonRetryingExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges()
   at HospitalAppTests.ServicesTests.PatientServiceTests.ClearDoctors(MyDbContext context) in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 249
   at HospitalAppTests.ServicesTests.PatientServiceTests.Set_random_general_practitioner() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 130
[xUnit.net 00:00:11.50]     HospitalAppTests.ServicesTests.PatientServiceTests.Add [FAIL]
[xUnit.net 00:00:11.54]     HospitalAppTests.ServicesTests.PatientServiceTests.Set_general_practitioner [FAIL]
  Failed HospitalAppTests.ServicesTests.PatientServiceTests.Add [81 ms]
  Error Message:
   System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
  Stack Trace:
     at System.Collections.Generic.List`1.get_Item(Int32 index)
   at HospitalApp.Services.PatientService.GiveRandomGeneralPractitioner(Patient patient) in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalApp\Services\PatientService.cs:line 61
   at HospitalApp.Services.PatientService.Add(Patient patient) in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalApp\Services\PatientService.cs:line 49
   at HospitalAppTests.ServicesTests.PatientServiceTests.Add() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 115
  Failed HospitalAppTests.ServicesTests.PatientServiceTests.Set_general_practitioner [32 ms]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at HospitalAppTests.ServicesTests.PatientServiceTests.Set_general_practitioner() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 65
Failed!  - Failed:     6, Passed:    60, Skipped:     0, Total:    66, Duration: 45 s - HospitalAppTests.dll (netcoreapp3.1)

Calculating coverage result...
One or more errors occurred. (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.)
